Bearded.UI/Controls/CompositeControl.cs

```diff
@@ -1,17 +1,22 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Linq;
 using Bearded.UI.Rendering;
+using Bearded.Utilities;
 
 namespace Bearded.UI.Controls
 {
-    public class CompositeControl : Control, IControlParent, IEnumerable<Control>
+    public class CompositeControl : Control, IControlParent, IFocusParent, IEnumerable<Control>
     {
         private readonly List<Control> children = new List<Control>();
 
         public ReadOnlyCollection<Control> Children { get; }
 
+        public bool HasFocusedDescendant => FocusState == FocusState.DescendantFocused;
+        private Maybe<Control> focusedDescendant = Maybe.Nothing;
```

Comment:
Maybe not something we're super concerned about, but since these are too different fields, we could potentially get into an invalid state more easily compared to it being a single one somehow. Since they are in two different classes, not sure if there's much we can do about it though?

I looked below for uses of focusedDescendant (in Blur), and I'm not sure why we actually need it? For the one use case I see we only need to have it stored in the root control somewhere which we can get by recursion, and then call Blur on that?